wandb/sdk/wandb_settings.py

```diff
@@ -710,7 +711,16 @@ def _default_props(self) -> Dict[str, Dict[str, Any]]:
             _offline={
                 "hook": (
                     lambda _: True
-                    if self.disabled or (self.mode in ("dryrun", "offline"))
+                    if self.disabled
+                    or (self.mode in ("dryrun", "offline", "offline_legacy"))
+                    else False
+                ),
+                "auto_hook": True,
+            },
+            _offline_legacy={
+                "hook": (
+                    lambda _: True
+                    if self.disabled or (self.mode == "offline_legacy")
```

Comment:
you don't need to worry about offline mode, the idea of this setting is to allow for users that know that their server supports client id to by pass the offline check, so the setting would something like `allow_for_offline_artifacts` or maybe a better name